pkg/workload/resource_test.go

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+	http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package workload
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	corev1 "k8s.io/api/core/v1"
+	nodev1 "k8s.io/api/node/v1"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	"sigs.k8s.io/kueue/pkg/controller/core/indexer"
+	utiltesting "sigs.k8s.io/kueue/pkg/util/testing"
+)
+
+func TestAdjustResources(t *testing.T) {
+	cases := map[string]struct {
+		runtimeClasses []nodev1.RuntimeClass
+		limitranges    []corev1.LimitRange
+		wl             *kueue.Workload
+		wantWl         *kueue.Workload
+	}{
+		"Handle Pod Overhead": {
+			runtimeClasses: []nodev1.RuntimeClass{
+				utiltesting.MakeRuntimeClass("runtime-a", "handler-a").
+					PodOverhead(corev1.ResourceList{
+						corev1.ResourceCPU:    ResourceQuantity(corev1.ResourceCPU, 1),
+						corev1.ResourceMemory: ResourceQuantity(corev1.ResourceMemory, 1024),
+					}).
+					RuntimeClass,
+			},
+			wl: utiltesting.MakeWorkload("foo", "").
+				PodSets(
+					*utiltesting.MakePodSet("a", 1).
+						RuntimeClass("runtime-a").
+						Obj(),
+				).
+				Obj(),
+			wantWl: utiltesting.MakeWorkload("foo", "").
+				PodSets(
+					*utiltesting.MakePodSet("a", 1).
+						RuntimeClass("runtime-a").
+						PodOverHead(
+							corev1.ResourceList{
+								corev1.ResourceCPU:    ResourceQuantity(corev1.ResourceCPU, 1),
+								corev1.ResourceMemory: ResourceQuantity(corev1.ResourceMemory, 1024),
+							}).
+						Obj(),
+				).
+				Obj(),
+		},
+		"Handle Pod Limit Range": {
+			limitranges: []corev1.LimitRange{
+				utiltesting.MakeLimitRange("foo", "").
+					WithType(corev1.LimitTypeContainer).
+					WithValue(
+						"Default", corev1.ResourceCPU, "4",
+					).
+					WithValue(
+						"DefaultRequest", corev1.ResourceCPU, "4",
+					).
+					WithValue(
+						"Max", corev1.ResourceCPU, "4",
```

Comment:
Can you make Max different with Default?

Reply:
OK,I was trying to simulate the condition that Default & DefaultRequest were not set,but it seems more resonable to set Max different from Default.